GameWorld/Camera/Camera.cs

```diff
@@ -0,0 +1,350 @@
+﻿using System;
+using Microsoft.Xna.Framework;
+using MonoGameWorld.Utilities;
+
+namespace MonoGameWorld.Camera
+{
+    public enum CameraType
+    {
+        Free = 0,
+        FirstPerson = 1,
+        ThirdPersonFree = 2,
+        ThirdPersonFreeAlt = 3,
+        ThirdPersonLocked = 4
+    }
+
+    class Camera
+    {
+        private CameraType cameraType;
+        private Vector3 offset;
+        private int width;
+        private int height;
+        private Quaternion rotation = Quaternion.Identity;
+        private Single fov;
+        private Vector3 up;
+        private Vector3? lookAt;
+        private Double radius;
+        
+        public CameraType CameraType
+        {
+            get { return cameraType; }
+            set
+            {
+                if (value != cameraType)
+                {
+                    {
+                        if ((cameraType == CameraType.ThirdPersonFree) || (cameraType == CameraType.ThirdPersonFreeAlt))
+                        {
+                            ViewMatrix.Decompose(out Vector3 scale, out rotation, out Vector3 translation);
+                        }
+                    }
+
+                    cameraType = value;
+                }
+            }
+        }
+        public Matrix ViewMatrix { get; set; } = Matrix.Identity;
+        public Matrix ProjectionMatrix { get; set; }
+        public Vector3 Offset
+        {
+            get { return offset; }
+            set
+            {
+                if (offset == value)
+                    return;
+
+                offset = value;
+            }
+        }
+        public Quaternion Rotation
+        {
+            get { return rotation; }
+            set
+            {
+                if (rotation == value)
```

Comment:
Do we really need such checks? Isn't it simpler to just assign without checking?